File: scripts/verification/verify_connection.js
/**
 * TripSage Database Connection Verification Script
 *
 * This script verifies the connection to the Supabase database
 * and checks that the required tables exist.
 *
 * Usage: node verify_connection.js
 */

const { createClient } = require("@supabase/supabase-js");
require("dotenv").config();

// Check for required environment variables
const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error(
    "ERROR: Missing environment variables SUPABASE_URL and/or SUPABASE_ANON_KEY"
  );
  console.error(
    "Please create a .env file based on .env.example and add your Supabase credentials"
  );
  process.exit(1);
}

// Create Supabase client
const supabase = createClient(supabaseUrl, supabaseKey);

// Tables that should exist
const requiredTables = [
  "users",
  "trips",
  "flights",
  "accommodations",
  "transportation",
  "itinerary_items",
  "search_parameters",
  "price_history",
  "trip_notes",
  "saved_options",
  "trip_comparison",
];

// Verify connection and tables
async function verifyConnection() {
  console.log("Connecting to Supabase at:", supabaseUrl);

  try {
    // Test a simple query to verify connection
    const { data, error } = await supabase.from("trips").select("id").limit(1);

    if (error) throw error;

    console.log("✅ Successfully connected to Supabase!");

    // Verify tables
    console.log("\nChecking required tables:");

    // Get list of tables using system schema
    const { data: tables, error: tablesError } = await supabase.rpc(
      "get_tables"
    );

    if (tablesError) {
      console.log(
        "Unable to list tables using RPC. Using alternative method..."
      );

      // Alternative method to check tables individually
      for (const table of requiredTables) {
        try {
          const { data, error } = await supabase
            .from(table)
            .select("id")
            .limit(1);

          if (error && error.code === "42P01") {
            console.log(`❌ Table '${table}' does not exist`);
          } else {
            console.log(`✅ Table '${table}' exists`);
          }
        } catch (err) {
          console.log(`❓ Could not verify table '${table}': ${err.message}`);
        }
      }

      return;
    }

    // If we have table list, check required tables
    const tableNames = tables.map((t) => t.table_name);

    for (const table of requiredTables) {
      if (tableNames.includes(table)) {
        console.log(`✅ Table '${table}' exists`);
      } else {
        console.log(`❌ Table '${table}' does not exist`);
      }
    }

    console.log("\nDatabase verification complete!");
  } catch (err) {
    console.error("❌ Error connecting to Supabase:", err.message);
    process.exit(1);
  }
}

// Run the verification
verifyConnection();

File: scripts/verification/verify_connection.py
#!/usr/bin/env python
"""TripSage Database Connection Verification Script.

This script verifies the connection to the Supabase database
and checks that the required tables exist.

Usage: python verify_connection.py
"""

import asyncio
import sys

from tripsage_core.services.business.auth_service import get_supabase_client
from tripsage_core.utils.logging_utils import configure_logging


# Configure logging
logger = configure_logging(__name__)

# Tables that should exist
REQUIRED_TABLES = [
    "users",
    "trips",
    "flights",
    "accommodations",
    "transportation",
    "itinerary_items",
    "search_parameters",
    "price_history",
    "trip_notes",
    "saved_options",
    "trip_comparison",
]


async def verify_connection() -> None:
    """Verify connection to Supabase database and check required tables exist."""
    print("Connecting to Supabase...")

    try:
        # Get a Supabase client
        supabase = get_supabase_client()

        # Test a simple query to verify connection
        _response = supabase.table("trips").select("id").limit(1).execute()

        print("✅ Successfully connected to Supabase!")

        # Verify tables
        print("\nChecking required tables:")

        # Get list of tables using system schema
        table_response = None
        try:
            table_response = supabase.rpc("get_tables").execute()
        except Exception:  # noqa: BLE001
            print("Unable to list tables using RPC. Using alternative method...")

        if table_response and table_response.data:
            # If we have table list, check required tables
            table_names = [t["table_name"] for t in table_response.data]

            for table in REQUIRED_TABLES:
                if table in table_names:
                    print(f"✅ Table '{table}' exists")
                else:
                    print(f"❌ Table '{table}' does not exist")
        else:
            # Alternative method to check tables individually
            for table in REQUIRED_TABLES:
                try:
                    _test_response = (
                        supabase.table(table).select("id").limit(1).execute()
                    )
                    print(f"✅ Table '{table}' exists")
                except Exception as e:  # noqa: BLE001
                    if "42P01" in str(
                        e
                    ):  # PostgreSQL error code for table does not exist
                        print(f"❌ Table '{table}' does not exist")
                    else:
                        print(f"❓ Could not verify table '{table}': {e!s}")

        print("\nDatabase verification complete!")
    except Exception as e:  # noqa: BLE001
        print(f"❌ Error connecting to Supabase: {e!s}")
        sys.exit(1)


if __name__ == "__main__":
    asyncio.run(verify_connection())
